Dockerfile

```diff
@@ -1,11 +1,10 @@
-FROM node:14
+FROM node:16-alpine
 WORKDIR /usr/src/app
-# Copying these separately prevents node_modules
```

Comment:
Are you confident that this comment is no longer applicable, and that the following COPY commands are no longer needed?

It looks like @skomorokh added those lines in the initial commit https://github.com/folio-org/mod-graphql/commit/0df9dc812bd6430d29e1f40048ac855bf1aa106c -- Jason, can you remember what that was all about?

Reply:
Using this Dockerfile
```
FROM node:16-alpine
WORKDIR /usr/src/app
COPY package.json .
COPY yarn.lock .
RUN yarn install
```
and running
```
docker build -t x .
docker run --rm -it x sh
```
yields
```
total 332
drwxr-xr-x    1 root     root          4096 Feb  8 09:20 .
drwxr-xr-x    1 root     root          4096 Feb  8 09:19 ..
drwxr-xr-x  747 root     root         28672 Feb  8 09:21 node_modules
-rw-rw-r--    1 root     root          2149 Feb  8 09:20 package.json
-rw-rw-r--    1 root     root        294002 Feb  8 09:20 yarn.lock
```
Only node_modules is created, but this directory is not in the repository and therefore cannot been overwritten.

Using this Dockerfile
```
FROM node:16-alpine
WORKDIR /usr/src/app
COPY . .
RUN yarn install
```
and running
```
docker build -t x .
docker run --rm -it x sh
```
yields
```
total 476
drwxr-xr-x    1 root     root          4096 Feb  8 09:30 .
drwxr-xr-x    1 root     root          4096 Feb  8 09:19 ..
-rw-rw-r--    1 root     root            27 Jan 29  2019 .dockerignore
-rw-rw-r--    1 root     root           261 Feb 16  2021 .eslintrc
drwxrwxr-x    8 root     root          4096 Feb  8 09:27 .git
-rw-rw-r--    1 root     root            44 Jan 29  2019 .gitignore
-rw-rw-r--    1 root     root            36 Feb 16  2021 .travis.yml
-rw-rw-r--    1 root     root         15268 Nov 30 21:25 CHANGELOG.md
-rw-rw-r--    1 root     root            67 Feb  8 09:29 Dockerfile
-rw-rw-r--    1 root     root           101 Jan 29  2019 ExternalDeploymentDescriptor.json
-rw-rw-r--    1 root     root           319 Feb 16  2021 Jenkinsfile
-rw-rw-r--    1 root     root         11357 Jan 29  2019 LICENSE
-rw-rw-r--    1 root     root           550 Nov 30 21:25 ModuleDescriptor.json
-rw-rw-r--    1 root     root         15160 May 11  2021 README.md
-rw-rw-r--    1 root     root            32 Jan 29  2019 TenantAssociationDescriptor.json
drwxrwxr-x    3 root     root          4096 Mar 23  2021 create-schemas
drwxrwxr-x    2 root     root          4096 Feb 16  2021 doc
-rw-rw-r--    1 root     root          6554 Jan 29  2019 instances.graphql-query
-rw-rw-r--    1 root     root           874 Feb 16  2021 main.js
drwxr-xr-x  747 root     root         28672 Feb  8 09:30 node_modules
-rw-rw-r--    1 root     root          2149 Jan 13 22:53 package.json
-rw-rw-r--    1 root     root           308 Feb 16  2021 query.graphql
-rw-rw-r--    1 root     root         22460 Jan 29  2019 result.json
-rwxrwxr-x    1 root     root           578 Feb 16  2021 run
drwxrwxr-x    3 root     root          4096 May 11  2021 src
drwxrwxr-x    5 root     root          4096 May 11  2021 tests
-rw-rw-r--    1 root     root        294002 Jan 13 22:53 yarn.lock
```

The timestamp shows that nothing gets overwritten.

Reply:
`node_modules` in listed in https://github.com/folio-org/mod-graphql/blob/v1.9.0/.dockerignore so that `COPY . .` doesn't copy it into the Docker container.

Reply:
OK, but I don't understand how this addresses the meaning of the cryptic comment "Copying these separately prevents `node_modules` being reinstalled on other changes". What can be meant be "other" changes?

Still, the change you have made evidently works, so ¯\_(ツ)_/¯

Reply:
The comment
> Copying these separately prevents node_modules being reinstalled on other changes
doesn't apply to the version that is checked into the repository, or has been wrong in the first place. The .gitignore ensures that the container starts without a node_modules directory.